frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx

```diff
@@ -29,6 +30,16 @@ export function RefinementSingleSelect({
          ...otherProps,
          sortBy: useSortBy(otherProps),
       });
+   const onClickSingleSelect = (newSelected: RefinementListItem) => {
+      refine(newSelected.value);
+
+      const oldSelected: RefinementListItem | undefined = items.find(
+         (item) => item.isRefined
+      );
+      if (oldSelected && oldSelected !== newSelected) {
+         refine(oldSelected.value);
+      }
+   };
```

Comment:
Not sure how much of an optimization this is, but doesn't seem like a bad thing to do here.

```suggestion
   const onClickSingleSelect = React.useCallback((newSelected: RefinementListItem) => {
      refine(newSelected.value);

      const oldSelected: RefinementListItem | undefined = items.find(
         (item) => item.isRefined
      );
      if (oldSelected && oldSelected !== newSelected) {
         refine(oldSelected.value);
      }
   }, [items, refine]);
```